### PR TITLE
add paragraph on casting static arrays

### PR DESCRIPTION
**Pull request by WalterBright**
Missing documentation on existing behavior.

### CODE REVIEW

File: spec/expression.dd [hunk 1]
```diff
@@ -1091,6 +1091,32 @@ $(GNAME CastExpression):
...
+
+        ---
+        import core.stdc.stdio;
+
+        void main()
+        {
+            byte[16] b = 4;
+            int[4] ia = cast(int[4]) b;
+            foreach (i; ia)
+                printf("%x\n");
```
Comment (aG0aep6G) [inline spec/expression.dd:1109]:
```suggestion
                printf("%x\n", i);
```

File: spec/expression.dd [hunk 2]
```diff
@@ -1091,6 +1091,32 @@ $(GNAME CastExpression):
...
+            /* prints:
+               3030303
+               3030303
+               3030303
+               3030303
```
Comment (aG0aep6G) [inline spec/expression.dd:1114]: It prints "4040404", doesn't it?

**Review by aG0aep6G — CHANGES_REQUESTED** [review]

File: spec/expression.dd (hunk 2 above)
Comment (WalterBright) [reply spec/expression.dd:1114]: Thanks! That's what I get for typing a working program in again. Fixed.

Comment (thewilsonator) [reply spec/expression.dd:1114]: Doesn't look like it. Forgot to push?

Comment (WalterBright) [reply spec/expression.dd:1114]: see line 1106

Comment (thewilsonator) [reply spec/expression.dd:1114]: Ah.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**